src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -286,14 +288,142 @@ def create_certificate(client, vault_base_url, certificate_name, certificate_pol
 
 create_certificate.__doc__ = KeyVaultClient.create_certificate.__doc__
 
+def add_certificate_contact(client, vault_base_url, contact_email, contact_name=None,
+                            contact_phone=None):
+    """ Add a contact to the specified vault to receive notifications of certificate operations. """
+    from azure.cli.command_modules.keyvault.keyvaultclient.generated.models import \
+        (Contact, Contacts, KeyVaultErrorException)
+    try:
+        contacts = client.get_certificate_contacts(vault_base_url)
+    except KeyVaultErrorException:
+        contacts = Contacts([])
+    contact = Contact(contact_email, contact_name, contact_phone)
+    if any((x for x in contacts.contact_list if x.email_address == contact_email)):
+        raise CLIError("contact '{}' already exists".format(contact_email))
+    contacts.contact_list.append(contact)
+    return client.set_certificate_contacts(vault_base_url, contacts)
+
+def delete_certificate_contact(client, vault_base_url, contact_email):
+    """ Remove a certificate contact from the specified vault. """
+    from azure.cli.command_modules.keyvault.keyvaultclient.generated.models import \
+        (Contacts, KeyVaultErrorException)
+    contacts = client.get_certificate_contacts(vault_base_url).contact_list
```

Comment:
You will get a KeyVaultErrorException in this case, but the infrastructure handles it by wrapping it in a CLIError and giving you a "Contacts not found" message from the server, which I think is fine.

The reason for the try/except in add_certificate_contact is because if you add a contact and there are no contacts, you have to catch the error to create the empty list so you can add your contact to that.